internal/storage/metrics/cache/cache.go

```diff
@@ -74,20 +74,20 @@ func (m *MemStorage) Select(ctx context.Context, el metrics.Element) (metrics.El
 		if f, ok := m.mapa[el.ID].(float64); ok {
 			out.Value = &f
 		} else {
-			return metrics.Element{}, fmt.Errorf("metric not found")
+			return nil, fmt.Errorf("metric not found")
 		}
 	} else if el.MType == "counter" {
 		if c, ok := m.mapa[el.ID].(int64); ok {
 			out.Delta = &c
 		} else {
-			return metrics.Element{}, fmt.Errorf("metric not found")
+			return nil, fmt.Errorf("metric not found")
 		}
 	}
 
-	return out, nil
+	return &out, nil
 }
 
-func (m *MemStorage) SelectAll(ctx context.Context) ([]metrics.Element, error) {
+func (m *MemStorage) SelectAll(ctx context.Context) (*[]metrics.Element, error) {
```

Comment:
```suggestion
func (m *MemStorage) SelectAll(ctx context.Context) ([]*metrics.Element, error) {
```
Либо слайс из ссылок на структуру, либо слайс из структур. Сам слайс является ссылочным типом и ссылка на него бывает нужна очень и очень редко (только когда подчиненная функция должна изменять именно этот слайс). Для данной итерации не блокер, можно поменять потом.